### PR TITLE
Closes #1336 Feature Request: Replace smq_select() and sdg_select() with basket_select()

### LINKED ISSUE #1336: Feature Request: Replace `smq_select()` and `sdg_select()` with `basket_select()`
### Feature Idea

```
basket_select(
  name = NULL,
  id = NULL,
  scope = NULL,
  type =)
```
should be implemented replacing `smq_select()` and `sdg_select()`. The `type` argument expects a character scalar. It is passed to the company specific `get_terms()` function such that the function can determine which sort of basket is requested, SMQ, SDG, or something else like AEGT at Roche. The `scope` argument should accept `"BROAD"`, `"NARROW"`, and `NA_character_`.

`create_query_data()` should be updated accordingly, i.e.,
- `meddra_version` and `whodd_version` argument should be replaced by `version` and `get_smq_fun` and `get_sdg_fun` argument by `get_terms_fun`,
- the expected arguments for the function specified for `get_terms_fun` should be changed to `type`, `basket_select`, `version`, `keep_id`, and `temp_env`.

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by tamarasenior**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
- [ ] Update to all relevant roxygen headers and examples 
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/create_query_data.R [hunk 1]
```diff
@@ -887,48 +824,54 @@ assert_terms <- function(terms,
...
   out <- list(
     name = name,
     id = id,
-    scope = scope
+    scope = scope,
+    type = type
   )
-  class(out) <- c("smq_select", "source", "list")
-  validate_smq_select(out)
+  class(out) <- c("basket_select", "list")
```
Comment (bundfussr) [inline R/create_query_data.R:855]: Please do not remove the `source` class.

**Comment by bundfussr** [comment]
@tamarasenior , could you add "@devel" to the branch name, i.e., rename `1336_feature_request_basket_select` to `1336_feature_request_basket_select@devel`. Then the devel versions of admiral.test and admiraldev are used in the workflows.

To include the PR in the next admiral release we should finish the review until end of next week (25th November).
Do you think this is feasible?

**Comment by tamarasenior** [comment]
Hi @bundfussr 
I've implemented all recommended updates, but I'm not sure how to rename the branch without closing the ticket? 
I'm also getting the following error after the type updates that I'm struggling to resolve when running the tests
unused argument (type = definition[[j]]$type)

**Comment by bms63** [comment]
> Hi @bundfussr I've implemented all recommended updates, but I'm not sure how to rename the branch without closing the ticket? I'm also getting the following error after the type updates that I'm struggling to resolve when running the tests unused argument (type = definition[[j]]$type)

You can close the PR and then just rename your branch.  
image
and then restart the PR process.  You can link the closed PR in the new PR if you want to keep the discussion linked together.  

I think since the issue is tagged in this PR it will auto-tag in the new PR that this was discussed.  !!

**Comment by bundfussr** [comment]
@tamarasenior , I have added some comments regarding to the error to the new PR.

**Pull request closed without merging**